src/extensions/runtimes/common.ts

```diff
@@ -125,14 +111,192 @@ function extensionMatchesFilter(codebase: string, extensionId: string, filter: F
     return true;
   }
 
-  const idChunks = extensionId.split("-");
-  if (idChunks.length < filter.idChunks.length) {
-    return false;
+  // Extension instance ids are not nested. They are unique to a project.
+  // They are allowed to have hyphens, so in the functions filter this will be
+  // interpreted as nested chunks, so we join them again to get the original id.
+  const filterId = filter.idChunks.join("-");
+
+  return extensionId === filterId;
+}
+
+export function isTypescriptCodebase(codebaseDir: string) {
+  return fsutils.fileExistsSync(path.join(codebaseDir, "tsconfig.json"));
+}
+
+export async function writeFile(filePath: string, data: string, options: any) {
+  const shortFilePath = filePath.replace(process.cwd(), ".");
+  if (fsutils.fileExistsSync(filePath)) {
+    if (
+      await confirm({
+        message: `${shortFilePath} already exists. Overwite it?`,
+        nonInteractive: options.nonInteractive,
+        force: options.force,
+        default: false,
+      })
+    ) {
+      // overwrite
+      try {
+        await fs.promises.writeFile(filePath, data, { flag: "w" });
+        logLabeledBullet("extensions", `successfully wrote ${shortFilePath}`);
+      } catch (err) {
+        throw new FirebaseError(`Failed to write ${shortFilePath}:\n    ${err}`);
+      }
+    } else {
+      // don't overwrite
+      return;
+    }
+  } else {
+    // write new file
+    // Make sure the directories exist
+    await fs.promises
+      .mkdir(path.dirname(filePath), { recursive: true })
+      .then(async () => {
```

Comment:
This `.then()` should be unnecessary. Pretty sure this code works identically if you just delete it and replace the .catch with a standard try/catch

Reply:
done